### PR TITLE
test for wp_cache_set_last_changed

### PR DESCRIPTION
**Pull request by pbearne**
Trac ticket: https://core.trac.wordpress.org/ticket/59737

### CODE REVIEW

File: tests/phpunit/tests/functions/wpCacheSetLastChanged.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	 * @ticket 59737
+	 */
+	public function test_wp_cache_set_last_changed_action_is_called() {
+
+		add_action(
+			'wp_cache_set_last_changed',
+			function ( $group ) {
+				$this->assertSame( 'group_name', $group );
+			}
+		);
```
Comment (peterwilsoncc) [inline tests/phpunit/tests/functions/wpCacheSetLastChanged.php]: It's best to use `MockAction` for this, in this form if the action doesn't fire the test won't fail but instead throw a warning that it does not contain any tests.

This test contains some prior art you can work from (ignore the calls removing the actions, they're not needed)

https://github.com/WordPress/wordpress-develop/blob/f0d53f8100d5bc274b126995e5f08a509a70d91c/tests/phpunit/tests/post.php#L395-L421

Comment (pbearne) [reply tests/phpunit/tests/functions/wpCacheSetLastChanged.php]: Thanks updated

File: tests/phpunit/tests/functions/wpCacheSetLastChanged.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * Tests for the wp_cache_set_last_changed function.
+ *
+ * @group functions.php
```
Comment (peterwilsoncc) [inline tests/phpunit/tests/functions/wpCacheSetLastChanged.php]:
```suggestion
 * @group functions
```

File: tests/phpunit/tests/functions/wpCacheSetLastChanged.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ * Tests for the wp_cache_set_last_changed function.
+ *
+ * @group functions
+ *
+ * @covers ::wp_cache_set_last_changed
+ */#
+class Tests_functions_wpCacheSetLastChanged extends WP_UnitTestCase {
+
+	/**
+	 * check the cache_set_last_changed is set.
```
Comment (peterwilsoncc) [inline tests/phpunit/tests/functions/wpCacheSetLastChanged.php]:
```suggestion
	 * Check the cache key last_changed is set for the specified group.
```

File: tests/phpunit/tests/functions/wpCacheSetLastChanged.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * Tests for the wp_cache_set_last_changed function.
+ *
+ * @group functions
+ *
+ * @covers ::wp_cache_set_last_changed
+ */#
```
Comment (peterwilsoncc) [inline tests/phpunit/tests/functions/wpCacheSetLastChanged.php]:
```suggestion
 */
```

File: tests/phpunit/tests/functions/wpCacheSetLastChanged.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * Tests for the wp_cache_set_last_changed function.
```
Comment (peterwilsoncc) [inline tests/phpunit/tests/functions/wpCacheSetLastChanged.php]:
```suggestion
 * Tests for the wp_cache_set_last_changed() function.
```

**Comment by peterwilsoncc** [comment]
Merged [r57060](https://core.trac.wordpress.org/changeset/57060)  / https://github.com/WordPress/wordpress-develop/commit/75796f5e92a73e46c07d05e86293dc4f878a15ed.

Thanks Paul!

**Pull request closed without merging**